### PR TITLE
Reconstruct BlobSidecars in EIP-7594

### PR DESCRIPTION
**Pull request by zilm13**
Partially addresses #98. Makes blobsidecars available when our node is supernode.
Also mimics deneb in more APIs because Kurtosis Dora starts breaking with current build.

### CODE REVIEW

File: storage/src/main/java/tech/pegasys/teku/storage/client/BlobSidecarReconstructionProvider.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                    .getMessage()
+                    .getBody()
+                    .getOptionalBlobKzgCommitments()
+                    .orElseThrow()
+                    .get(blobIndex)
+                    .getKZGCommitment())));
+  }
+
+  public SafeFuture<Optional<SignedBeaconBlock>> getBlockByBlockRoot(final Bytes32 blockRoot) {
+    return recentChainData
```
Comment (Nashatyrev) [inline storage/src/main/java/tech/pegasys/teku/storage/client/BlobSidecarReconstructionProvider.java]: Isn't this logic of the `CombinedChainDataClient`

File: storage/src/main/java/tech/pegasys/teku/storage/client/CombinedChainDataClient.java [hunk 2]
```diff
@@ -173,6 +178,18 @@ public SafeFuture<Optional<SignedBlockAndState>> getSignedBlockAndStateInEffectA
    */
   public SafeFuture<List<BlobSidecar>> getBlobSidecars(
       final UInt64 slot, final List<UInt64> indices) {
+    if (spec.atSlot(slot).getMilestone().isGreaterThanOrEqualTo(SpecMilestone.EIP7594)) {
+      // TODO: suboptimal
```
Comment (Nashatyrev) [inline storage/src/main/java/tech/pegasys/teku/storage/client/CombinedChainDataClient.java]: Do you think we need incorporate this complexity to the `CombinedChainDataClient` ? 
May be pull to the upper level classes?

**Comment by zilm13** [comment]
refactored + tests added

**Review by Nashatyrev — APPROVED** [review]
LGTM

**Pull request merged**